### ISSUE TITLE
Standalone settings are not used when analysing JS or TS when in connected mode

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#2453]
### Description

Neither TS nor JS support connected mode yet, so the standalone settings should be used to allow users to disable noisy rules.
However, the standalone settings are not used.

### Repro
1. Open a bound solution
2. Add a JS or TS file containing an issue
3. Disable the rule

### Expected
The file is reanalysed and the issue is not shown.

### Actual
The file is not reanalysed. When the file is re-analysed (e.g. on save), the issue is still reported.

```
Disabled rule "typescript:S1751"
File changed: C:\Users\xxx\AppData\Roaming\SonarLint for Visual Studio\settings.json. Change type: Changed
Loaded settings from "C:\Users\xxx\AppData\Roaming\SonarLint for Visual Studio\settings.json".
User settings have changed but the solution is in connected mode. Open documents will not be re-analysed.
```

### Notes

`AnalysisConfigMonitor.OnUserSettingsChanged` does not request re-analysis in connected mode, on the assumption that the local settings are not relevant. However, JS and TS don't currently support Connected Mode so that assumption is incorrect.

Triggering re-analysis in Connected Mode should not cause a problem:
* the reanalysis should return the same issues.
* the `Disable Rule` command will only be enabled for JS and TS, and if a is rule is disabled we want the reanalysis to occur in this case
* the only way other way the user settings file could change is if the user edits it directly. We wouldn't expect this to happen often in Connected Mode.

It might be worth logging a message in `AnalysisConfigMonitor.OnUserSettingsChanged` when in Connected Mode saying that the local settings will be ignored for languages that support Connected Mode.

Affected SonarLint version(s): v4.34+
Language(s): JavaScript (will affect TypeScript when v4.35 is released)
